src/wp-includes/blocks.php

```diff
@@ -723,29 +723,28 @@ function get_dynamic_block_names() {
 }
 
 /**
- * Retrieves block types (and positions) hooked into the given block.
+ * Retrieves block types hooked into the given block, grouped by their relative position.
  *
  * @since 6.4.0
  *
- * @param string $name              Block type name including namespace.
- * @param string $relative_position Optional. Relative position of the hooked block. Default empty string.
- * @return array Associative array of `$block_type_name => $position` pairs.
+ * @param string $name Block type name including namespace.
+ * @return array[] Array of block types grouped by their relative position.
  */
-function get_hooked_blocks( $name, $relative_position = '' ) {
+function get_hooked_blocks( $name ) {
 	$block_types   = WP_Block_Type_Registry::get_instance()->get_all_registered();
 	$hooked_blocks = array();
```

Comment:
Maybe, we could avoid all the further conditional checks by initializing all possible relative positions?

```php
$hooked_blocks = array(
	'before'.      => array(),
	'after'          => array(),
	'first_child' => array(),
	'last_child'  => array(),
);
```

What do you think?

Reply:
See [my other comment](https://github.com/WordPress/wordpress-develop/pull/5297#discussion_r1335955703) (which was probably hidden after rebasing):

> The alternative would be to make `get_hooked_blocks` create empty arrays for all positions, but I didn't like that option so much either; one drawback would be that we'd need to give `get_hooked_blocks` a list of possible relative positions (which we'd also need to update if we introduce new ones); whereas with the current approach, we can keep `get_hooked_blocks` ignorant of what relative positions exist, and let it create arrays for whatever it finds.

Reply:
I missed that comment somehow.

It's also possible to put allowed values in a constant on `WP_Block_Type`:

```php
const BLOCK_HOOKS_POSITIONS = array( 'after', 'before', 'first_child', 'last_child' );
```

This way, we don't need to worry about the maintenance if we use that constant to initialize the `$hooked_block` array.

Reply:
Yeah, I was also considering a const, but still not totally loving the idea 😬 

I kinda prefer a return value of e.g.

```php
array(
	'first_child' => array(
		'tests/injected-two',
	),
)
```

to all the empty arrays in

```php
array(
	'before'      => array(),
	'after'       => array(),
	'first_child' => array(
		'tests/injected-two',
	),
	'last_child'  => array(),
)
```

which is kinda noisy. But again, that might be a matter of taste 😊